### PR TITLE
[7758] AStyle coding style tool added as part of RTR tool

### LINKED ISSUE #7758: Add ASTYLE to cpp wazuh projects
## Description
This issue aims to be able to add a code style checker, this with enough configuration, will automatically help to respect certain standards defined in the project's coding style.

In this way, do not open new code that does not comply with what is defined.

URL:
http://astyle.sourceforge.net/

Source branch: Master
Destination branch: Master

## DoD
- [x] Create custom configuration based on the coding style guideline.
- [x] Add to RTR diff check to set red or green flag.
- [x] Request add astyle as dependency on rtr docker file.
Issue created in wazuh-jenkins repo: https://github.com/wazuh/wazuh-jenkins/issues/2419
- [x] PR CI Pass.

### PR DESCRIPTION
**Pull request by tinachoB**
|Related issue|
|---|
|Closes #7758 |

## **Description**
This issue aims to be able to add a code style checker, this with enough configuration, will automatically help to follow certain standards defined in the project's [C++ coding style](https://github.com/wazuh/wazuh/wiki/Coding-style-guide-(C-plus-plus)).

In this way, do not open new code that does not comply with what is defined.

**URL:** http://astyle.sourceforge.net/

**Wazuh-jenkins related issue:** https://github.com/wazuh/wazuh-jenkins/issues/2419

## **DoD**
- [X] Update of rtr tool
- [X] Update Readme.md with new options
- [X] RTR tool execution with new flag
image
- [X] CI PASS

## Tests

- [x] Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [X] Scan-build report
  
image

  - [ ] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [x] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

### CODE REVIEW

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/data_provider/src/ports/portLinuxWrapper.h [hunk 1]
```diff
@@ -16,8 +16,14 @@
 #include "sharedDefs.h"
 #include "bits/stdc++.h"
 
-constexpr auto IPv6AddressHexSize { 32 };
-constexpr auto IPv4AddressHexSize { 8 };
+constexpr auto IPv6AddressHexSize
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/data_provider/src/ports/portLinuxWrapper.h:19]: Will do

File: src/data_provider/src/sysInfoMac.cpp [hunk 2]
```diff
@@ -340,15 +366,17 @@ nlohmann::json SysInfo::getPorts() const
         {
             nlohmann::json port;
             std::make_unique<PortImpl>(std::make_shared<BSDPortWrapper>(processInfo.first, fdSocket))->buildPortData(port);
+
             if (ports["ports"].end() == std::find_if(ports["ports"].begin(), ports["ports"].end(),
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/data_provider/src/sysInfoMac.cpp:370]: Yes. Will fix it in my next commit.

File: src/shared_modules/dbsync/tests/interface/dbsync_test.cpp [hunk 3]
```diff
@@ -238,7 +238,7 @@ TEST_F(DBSyncTest, dbsyncAddTableRelationship)
                 }
             ]
         })"
-    };
+                               };
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/dbsync/tests/interface/dbsync_test.cpp:241]: Yes. Will fix it in my next commit.

File: src/shared_modules/dbsync/tests/interface/dbsync_test.cpp [hunk 4]
```diff
@@ -1731,7 +1731,7 @@ TEST_F(DBSyncTest, dbsyncAddTableRelationshipCPP)
...
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/dbsync/tests/interface/dbsync_test.cpp:1734]: Yes. Will fix it in my next commit.

File: src/shared_modules/dbsync/tests/pipelineFactory/dbsyncPipelineFactory_test.cpp [hunk 5]
```diff
@@ -70,8 +70,8 @@ TEST_F(DBSyncPipelineFactoryTest, CreatePipelineInvalidHandle)
                                  json["tables"],
                                  threadNumber,
                                  maxQueueSize,
-                                 [](ReturnTypeCallback, const nlohmann::json&){}),
-        DbSync::dbsync_error
+    [](ReturnTypeCallback, const nlohmann::json&) {}),
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/dbsync/tests/pipelineFactory/dbsyncPipelineFactory_test.cpp:73]: Yes. Will fix it in my next commit.

File: src/shared_modules/dbsync/tests/pipelineFactory/dbsyncPipelineFactory_test.cpp [hunk 6]
```diff
@@ -86,8 +86,8 @@ TEST_F(DBSyncPipelineFactoryTest, CreatePipelineInvalidTxnContext)
...
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/dbsync/tests/pipelineFactory/dbsyncPipelineFactory_test.cpp:89]: Yes. Will fix it in my next commit.

File: src/shared_modules/utils/tests/pipelineNodes_test.cpp [hunk 7]
```diff
@@ -62,21 +65,24 @@ TEST_F(PipelineNodesTest, ReadWriteNodeAsync)
     {
         std::make_shared<ReadWriteNode>
         (
-            [](const std::string& value)
-            {
-                return std::stoi(value);
-            }
+            [](const std::string & value)
+        {
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/utils/tests/pipelineNodes_test.cpp:69]: Yes. Will fix it in my next commit.

File: src/data_provider/src/sysInfoWin.cpp [hunk 8]
```diff
@@ -38,191 +38,209 @@
 #include "ports/portImpl.h"
 #include "packages/packagesWindowsParserHelper.h"
 
-constexpr auto BASEBOARD_INFORMATION_TYPE{2};
-constexpr auto CENTRAL_PROCESSOR_REGISTRY{"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0"};
+constexpr auto BASEBOARD_INFORMATION_TYPE
+{
+    2
```
Comment (Dwordcito) [inline src/data_provider/src/sysInfoWin.cpp:43]: this could be one line constexpr ?

File: src/shared_modules/dbsync/include/db_exception.h [hunk 9]
```diff
@@ -14,25 +14,82 @@
...
-constexpr auto INVALID_TABLE                  { std::make_pair(13, "Invalid table.") };
-constexpr auto INVALID_DELETE_INFO            { std::make_pair(14, "Invalid information provided for deletion.") };
-constexpr auto BIND_FIELDS_DOES_NOT_MATCH     { std::make_pair(15, "Invalid information provided for statement creation.") };
-constexpr auto STEP_ERROR_CREATE_STMT         { std::make_pair(16, "Error creating table.") };
-constexpr auto STEP_ERROR_ADD_STATUS_FIELD    { std::make_pair(17, "Error adding status field.") };
-constexpr auto STEP_ERROR_UPDATE_STATUS_FIELD { std::make_pair(18, "Error updating status field.") };
-constexpr auto STEP_ERROR_DELETE_STATUS_FIELD { std::make_pair(19, "Error deleting status field.") };
+constexpr auto FACTORY_INSTANTATION
+{
+    std::make_pair(1, "Unspecified type during factory instantiation")
```
Comment (Dwordcito) [inline src/shared_modules/dbsync/include/db_exception.h:19]: this could be one line consexpr ?

File: src/data_provider/src/sysInfoWin.cpp (hunk 8 above)
Comment (tinachoB) [reply src/data_provider/src/sysInfoWin.cpp:43]: Yes. I'll do it

File: src/shared_modules/dbsync/include/db_exception.h (hunk 9 above)
Comment (tinachoB) [reply src/shared_modules/dbsync/include/db_exception.h:19]: Yes. I'll do it

**Comment by tinachoB** [comment]
@Dwordcito  - Please re check this. Thanks!

**Review by konangc87 — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**